src/contracts/core/AllocationManager.sol

```diff
@@ -313,101 +285,49 @@ contract AllocationManager is
             );
 
             bytes32 operatorSetKey = _encodeOperatorSet(allocation.operatorSets[i]);
-            // prep magnitudeUpdates in storage
-
-            // Check that there is at MOST `MAX_PENDING_UPDATES` combined allocations & deallocations for the operator, operatorSet, strategy
-            (uint256 numPendingAllocations, uint256 numPendingDeallocations) =
-                _getPendingAllocationsAndDeallocations(operator, allocation.strategy, operatorSetKey);
 
+            // Check that there are no pending allocations & deallocations for the operator, operatorSet, strategy
+            MagnitudeInfo memory opsetMagnitudeInfo = _operatorMagnitudeInfo[operator][allocation.strategy][operatorSetKey];
             require(
-                numPendingAllocations + numPendingDeallocations < MAX_PENDING_UPDATES,
+                opsetMagnitudeInfo.effectTimestamp <= block.timestamp,
                 PendingAllocationOrDeallocation()
             );
 
-            Snapshots.History storage magnitudeUpdates = _magnitudeUpdate[operator][allocation.strategy][operatorSetKey];
-
-            // Read current magnitude allocation including its respective array index and length.
-            // We'll use these values later to check the number of pending allocations/deallocations.
-            (uint224 currentMagnitude,,) =
-                magnitudeUpdates.upperLookupRecentWithPos(uint32(block.timestamp));
+            // Set the current magnitude to the pending magnitude if the effectTimestamp is in the past
+            uint64 currentMagnitude = _calculateNewCurrentMagnitude(opsetMagnitudeInfo);
```

Comment:
Is this comment correct? If the effectTimestamp is not in the past we just revert - are we supposed to be handling it somehow?